### PR TITLE
Prepend table name with owner's username.

### LINKED ISSUE #9516: Prefix sql query with user when the dataset is shared inside org
If an owner is working with a dataset, the default query (or custom) doesn't include the user prefix for tables' names. If this user shares the dataset with another user, the data visualisation fails because `the relation doesn't exist`. It's necessary to prefix the table's name with the owner's name in order to work properly.

~~Also, there is an inconsistency between the view when the dataset is shared with read permission and write and read permission. In the first scenario, the user is redirected to the public view.~~

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #9516.

It prepends the owner's username to table name if the dataset it's shared.

cc @xavijam

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/helpers/sql-utils.js [hunk 1]
```diff
@@ -60,6 +60,18 @@ module.exports = {
 
   _removeLineBreaks: function (sql) {
     return sql.replace(/\r?\n|\r/g, ' ');
-  }
+  },
+
+  prependTableName: function (sql, tableModel, userModel) {
+    var isOwner = tableModel.isOwner(userModel);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/helpers/sql-utils.js]: Not sure about having this logic in this class. For me we should prepend the user name before the table is needed, but we should check if it is necessary before using it.

Comment (nobuti) [inline lib/assets/javascripts/cartodb3/helpers/sql-utils.js]: Do you mean in `tableModel::getUnquotedName()`? But in any case, we have to make a replace in the query that comes from the backend right? How does it solve this case?

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/helpers/sql-utils.js]: Yeah, replace is needed, and I like the idea of having a method to do that here. But I would not check if the action is needed here, I'd do it before.

File: lib/assets/javascripts/cartodb3/dataset.js [hunk 2]
```diff
@@ -73,8 +74,10 @@ UserGroupFetcher.track({
   acl: visModel.getPermissionModel().acl
 });
 
+var query = SQLUtils.prependTableName(layerModel.get('sql') || 'SELECT * FROM ' + Utils.safeTableNameQuoting(tableModel.getUnquotedName()), tableModel, userModel);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/dataset.js]: Basically here, for example, check if sql has to be qualified.

File: lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-options-view.js [hunk 3]
```diff
@@ -275,7 +275,8 @@ module.exports = CoreView.extend({
   },
 
   _defaultSQL: function () {
-    return 'SELECT * FROM ' + this._tableModel.getUnquotedName();
+    var defaultQuery = 'SELECT * FROM ' + this._tableModel.getUnquotedName();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-options-view.js]: And here.

File: lib/assets/javascripts/cartodb3/dataset.js (hunk 2 above)
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/dataset.js]: 👍  It's more or less how I had it initially, but in order to not repeat the `isOwner` condition calculation every time the method is called , I moved it inside the helper.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/dataset.js]: It is only in two places, not that bad. I wanted to have sql-utils kind of easy and simple and not knowing about rest of the things outside.

**Comment by xavijam** [comment]
I feel we need it in the builder view too.

**Comment by xavijam** [comment]
:kiss:

**Comment by xavijam** [comment]
Come on!

**Pull request merged**